src/niscope/metadata/functions_addon.py

```diff
@@ -4,4 +4,146 @@
 functions_override_metadata = {
 }
 
+functions_additional_get_self_cal_last_date_and_time = {
+    'FancyGetLastSelfCalDateAndTime': {
+        'codegen_method': 'python-only',
+        'documentation': {
+            'description': 'Returns the date and time of the last self calibration performed.',
+        },
+        'method_templates': [
+            {
+                'documentation_filename': 'default_method',
+                'method_python_name_suffix': '',
+                'session_filename': 'fancy_cal_last_date'
+            }
+        ],
+        'parameters': [
+            {
+                'direction': 'in',
+                'documentation': {
+                    'description': 'Identifies a particular instrument session. You obtain the **vi** parameter from niScope_init.'
+                },
+                'name': 'vi',
+                'type': 'ViSession'
+            },
+            {
+                'direction': 'out',
+                'documentation': {
+                    'description': 'Indicates the **datetime** of the last calibration.'
+                },
+                'name': 'datetime',
+                'type': 'ViInt32'
+            }
+        ],
+        'python_name': 'get_self_cal_last_date_and_time',
+        'returns': 'ViStatus'
+    }
+}
+
+functions_additional_get_ext_cal_last_date_and_time = {
+    'FancyGetLastExtCalDateAndTime': {
+        'codegen_method': 'python-only',
+        'documentation': {
+            'description': 'Returns the date and time of the last external calibration performed.',
+        },
+        'method_templates': [
+            {
+                'documentation_filename': 'default_method',
+                'method_python_name_suffix': '',
+                'session_filename': 'fancy_cal_last_date'
+            }
+        ],
+        'parameters': [
+            {
+                'direction': 'in',
+                'documentation': {
+                    'description': 'Identifies a particular instrument session. You obtain the **vi** parameter from niScope_init.'
+                },
+                'name': 'vi',
+                'type': 'ViSession'
+            },
+            {
+                'direction': 'out',
+                'documentation': {
+                    'description': 'Indicates the **datetime** of the last calibration.'
+                },
+                'name': 'datetime',
+                'type': 'ViInt32'
+            }
+        ],
+        'python_name': 'get_ext_cal_last_date_and_time',
+        'returns': 'ViStatus'
+    }
+}
 
+functions_additional_get_self_cal_last_temp = {
+    'FancyGetLastSelfCalTemp': {
+        'codegen_method': 'python-only',
+        'python_name': 'get_self_cal_last_temp',
+        'documentation': {
+            'description': 'Returns the **Temperature** in degrees Celsius during the last self calibration procedure.',
```

Comment:
Where did you get the docstrings from?

The C function reference for `niScope_CalFetchTemperature` has more details. Python documentation should be as close to C function reference as possible,

Reply:
Changed to be in line with [official documentation](https://www.ni.com/documentation/en/ni-scope/20.0/scopenodes/get-last-calibration-temperature/). Since the internal methods for niScope_CalFetchTemperature and niScope_CalFetchDate had no generated documentation, I was previously referencing documentation for another module.